From b571713f51beb467eefae8e56bfea6e5eab48f8c Mon Sep 17 00:00:00 2001
From: James Feist <james.feist@linux.intel.com>
Date: Tue, 10 Dec 2019 16:21:24 -0800
Subject: [PATCH 1/1] AST2500 increase boot speed

Enable CONFIG_FLASH_SPIx2_Dummy and add it to known
good SPI flashes.

Signed-off-by: James Feist <james.feist@linux.intel.com>
---
 arch/arm/mach-aspeed/flash.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-aspeed/flash.c b/arch/arm/mach-aspeed/flash.c
index d33fb9e0fe..95fc46dd76 100644
--- a/arch/arm/mach-aspeed/flash.c
+++ b/arch/arm/mach-aspeed/flash.c
@@ -34,6 +34,7 @@
 #include <asm/arch/ast_scu.h>
 #include <asm/arch/aspeed.h>
 
+#define CONFIG_FLASH_SPIx2_Dummy 1
 
 /*
  * This file implements a Common Flash Interface (CFI) driver for U-Boot.
@@ -730,7 +731,7 @@ static ulong flash_get_size (ulong base, flash_info_t *info)
 	ulID = ((ulong)ch[0]) | ((ulong)ch[1] << 8) | ((ulong)ch[2] << 16) ;
 	info->flash_id = ulID;
 
-//	printf("SPI Flash ID: %x \n", ulID);
+	printf("SPI Flash ID: %x \n", ulID);
 
 	/* init default */
 	info->iomode = IOMODEx1;
@@ -1035,6 +1036,12 @@ static ulong flash_get_size (ulong base, flash_info_t *info)
 			EraseClk = 25;
 			ReadClk  = 50;
 			info->address32 = 1;
+#if	defined(CONFIG_FLASH_SPIx2_Dummy)
+			info->readcmd = 0xbb;
+			info->dummybyte = 1;
+			info->dualport = 1;
+			info->iomode = IOMODEx2_dummy;
+#endif
 			break;
 
 		case SST25VF016B:
-- 
2.17.1

